tests/integration/test_integration_authenticator_state.py

```diff
@@ -0,0 +1,90 @@
+
+    async def test_authenticator_enable_success(self):
+        c = await self._create_client(self.valid_user.id, self.valid_user.api_key)
+        response = await c.set_authenticator_state('authn-iam/test', True)
+
+        self.assertEqual(response, '')
+
+    async def test_authenticator_disable_success(self):
+        c = await self._create_client(self.valid_user.id, self.valid_user.api_key)
+        response = await c.set_authenticator_state('authn-iam/test', False)
+
+        self.assertEqual(response, '')
+
+    async def test_authenticator_enable_no_permissions(self):
+        c = await self._create_client(self.invalid_user.id, self.invalid_user.api_key)
+
+        with self.assertRaises(HttpStatusError) as context:
+            response = await c.set_authenticator_state('authn-iam/test', True)
+        self.assertEqual(context.exception.status, 403)
+
+    async def test_authenticator_disable_no_permissions(self):
+        c = await self._create_client(self.invalid_user.id, self.invalid_user.api_key)
+        with self.assertRaises(HttpStatusError) as context:
+            response = await c.set_authenticator_state('authn-iam/test', False)
+        self.assertEqual(context.exception.status, 403)
+
+    async def test_enable_non_existing_authenticator(self):
+        c = await self._create_client(self.valid_user.id, self.valid_user.api_key)
+
+        with self.assertRaises(HttpStatusError) as context:
+            response = await c.set_authenticator_state('authn-iam/non-existing', True)
+        self.assertEqual(context.exception.status, 401)
+
+    async def test_disable_non_existing_authenticator(self):
+        c = await self._create_client(self.valid_user.id, self.valid_user.api_key)
+        with self.assertRaises(HttpStatusError) as context:
+            response = await c.set_authenticator_state('authn-iam/non-existing', False)
+        self.assertEqual(context.exception.status, 401)
+
+    @classmethod
+    async def _add_test_data(cls):
+        c = await cls._create_client("admin", os.environ['CONJUR_AUTHN_API_KEY'])
+        response = await c.load_policy_file('root', 'tests/integration/policies/root.yml')
+        valid_user: ConjurUser = ConjurUser(user_id='test-authenticator-valid-user',
+                                            api_key=response['created_roles']['dev:user:test-authenticator-valid-user']
+                                            ['api_key'])
+        invalid_user: ConjurUser = ConjurUser(user_id='test-authenticator-invalid-user',
+                                              api_key=response['created_roles']['dev:user:test-authenticator-invalid-user']
+                                              ['api_key'])
+        response = await c.load_policy_file('conjur', 'tests/integration/policies/authn-iam-test.yml')
+
+        cls.valid_user = valid_user
+        cls.invalid_user = invalid_user
+
+    @classmethod
+    async def _create_client(cls, username: str, password: str) -> Client:
```

Comment:
Just for understanding: does that function create real SDK client, but it does not interact with real conjur?
How does this magic happen?

Reply:
It is interacting with a real Conjur. The integration test runs a local Conjur environment and then runs the tests against it. See the `ubuntu_compose.yml` file located in `ci/test/conjur-deployment` for more details

Reply:
Ohh now I understand :)